src/NSwag.AspNetCore/IDocumentProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IDocumentProvider.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.ApiDescription
+{
+    // This service will be looked up by name from the service collection when using
+    // the Microsoft.Extensions.ApiDescription tool
+    public interface IDocumentProvider
```

Comment:
So this interface is the only public API which is relevant for you, is this correct? 

Shouldn't it be renamed to IOpenApiDocumentProvider or also use a document type enum - because there might be other document types in the future....

Reply:
This doesn't need to be public

Reply:
This is a contract that we're implementing via late binding - it's not going to have types like enums - because there's no library. It's a way for us to consume a document in a generic way.

If you want a different abstraction for inside NSwag that sounds good, but it's not what we will consume.

Reply:
Ok, if we can make it internal and it doesnt give name collisions problems with other libraries (e.g. Swashbuckle in the same app) I think it's fine.